hooks/src/index.js

```diff
@@ -228,9 +228,14 @@ export function useDebugValue(value, formatter) {
 function flushAfterPaintEffects() {
 	afterPaintEffects.some(component => {
 		if (component._parentDom) {
-			component.__hooks._pendingEffects.forEach(invokeCleanup);
-			component.__hooks._pendingEffects.forEach(invokeEffect);
-			component.__hooks._pendingEffects = [];
+			try {
+				component.__hooks._pendingEffects.forEach(invokeCleanup);
+				component.__hooks._pendingEffects.forEach(invokeEffect);
```

Comment:
These callbacks are also invoked in the `_render` hook but in that case the exceptions are not caught.

Reply:
> For the first case we do a synchronous cleanup for the effects still pending for a component this can happen with many subsequent renders, here we use the _render option this is within the diffing loop so if a throw occurs it will catch it just as a normal throw in render would.

Reply:
as I understand it `_render` is handled [here](https://github.com/preactjs/preact/blob/master/src/diff/index.js#L217-L219). the main issue isn't that there is an exception, but that `afterPaintEffects` is needs to be mutated here for the next flush to happen. the other invocations don't suffer from the same affliction